### PR TITLE
[fix] legacybatching: getPhysicalDims

### LINKED ISSUE #92985: `torch.amax` will fail when computing forward mode jacobian
### 🐛 Describe the bug

`torch.amax` will fail when computing forward mode jacobian with error `RuntimeError: Could not allocate memory to change Tensor SizesAndStrides! `. But its reverse mode jacobian can be computed without any error.

The forward mode jacobian computation
```py
input = torch.rand([2], dtype=torch.float32)

def func(input):
    res = torch.amax(input)
    return res

jacobian(func, (input.requires_grad_(), ), strategy='forward-mode', vectorize=True)
# RuntimeError: Could not allocate memory to change Tensor SizesAndStrides!
```

For the reverse mode,
```py
input = torch.rand([2], dtype=torch.float32)

def func(input):
    res = torch.amax(input)
    return res

jacobian(func, (input.requires_grad_(), ), strategy='reverse-mode', vectorize=True)
# (tensor([0., 1.]),)
```

### Versions

```
PyTorch version: 2.0.0.dev20230105
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.1 LTS (x86_64)
GCC version: (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
Clang version: Could not collect
CMake version: version 3.22.1
Libc version: glibc-2.35

Python version: 3.9.15 (main, Nov 24 2022, 14:31:59)  [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.15.0-56-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: 11.7.99
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration:
GPU 0: NVIDIA GeForce RTX 3090
GPU 1: NVIDIA GeForce RTX 3090
GPU 2: NVIDIA GeForce RTX 3090

Nvidia driver version: 515.86.01
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.4.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.4.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.4.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.4.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.4.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.4.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.4.1
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.23.5
[pip3] torch==2.0.0.dev20230105
[pip3] torchaudio==2.0.0.dev20230105
[pip3] torchvision==0.15.0.dev20230105
[conda] blas                      1.0                         mkl
[conda] mkl                       2021.4.0           h06a4308_640
[conda] mkl-service               2.4.0            py39h7f8727e_0
[conda] mkl_fft                   1.3.1            py39hd3c417c_0
[conda] mkl_random                1.2.2            py39h51133e4_0
[conda] numpy                     1.23.5           py39h14f4228_0
[conda] numpy-base                1.23.5           py39h31eccc5_0
[conda] pytorch                   2.0.0.dev20230105 py3.9_cuda11.7_cudnn8.5.0_0    pytorch-nightly
[conda] pytorch-cuda              11.7                 h67b0de4_2    pytorch-nightly
[conda] pytorch-mutex             1.0                        cuda    pytorch-nightly
[conda] torchaudio                2.0.0.dev20230105      py39_cu117    pytorch-nightly
[conda] torchtriton               2.0.0+0d7e753227            py39    pytorch-nightly
[conda] torchvision               0.15.0.dev20230105      py39_cu117    pytorch-nightly
```

cc @ezyang @gchanan @zou3519 @Chillee @samdow @soumith @kshitij12345 @janeyx99

### PR DESCRIPTION
**Pull request by kshitij12345**
Fixes #92985

Minimum Repro:
```python
import torch
from torch._vmap_internals import vmap

input = torch.randn(2, 2)

def fn(x):
    return x.sum(())

o = vmap(fn)(input)
```

### CODE REVIEW

**Review by albanD — APPROVED** [review]
Perfect, thanks!

File: aten/src/ATen/LegacyVmapTransforms.cpp [hunk 1]
```diff
@@ -61,7 +61,7 @@ VmapDimVector VmapPhysicalView::getPhysicalDims(OptionalIntArrayRef opt_logical_
   // NB: fmap doesn't have a SmallVector variant, so we don't use it here.
   VmapDimVector result;
   result.reserve(logical_ndim);
-  if (opt_logical_dims.has_value()) {
+  if (opt_logical_dims.has_value() && opt_logical_dims.value().size() > 0) {
```
Comment (Skylion007) [inline aten/src/ATen/LegacyVmapTransforms.cpp:64]:
```suggestion
  if (opt_logical_dims.has_value() && !opt_logical_dims.value().empty()) {
```

Comment (kshitij12345) [reply aten/src/ATen/LegacyVmapTransforms.cpp]: Cool. Thanks!

**Review by Skylion007 — APPROVED** [review]

**Pull request closed without merging**